packages/origin-247-certificate/src/offchain-certificate/certificate.aggregate.ts

```diff
@@ -0,0 +1,188 @@
+import {
+    ICertificateEvent,
+    CertificateEventType,
+    CertificateTransferredEvent,
+    CertificateClaimedEvent
+} from './events/Certificate.events';
+import { IIssueCommand, ICertificateReadModel } from '../types';
+import { CertificateErrors } from './errors';
+import { BigNumber } from 'ethers';
+
+interface ConstructorParams {}
+interface FromEvents extends ConstructorParams {
+    events: ICertificateEvent[];
+    cert: never;
+}
+interface FromReadModel extends ConstructorParams {
+    cert: ICertificateReadModel;
+    events: never;
+}
+export class CertificateAggregate {
+    private certificate: ICertificateReadModel | null = null;
+
+    private constructor(params: FromEvents | FromReadModel) {
+        if (params.events) {
+            params.events
+                .sort((first, second) => {
+                    return first.internalCertificateId - second.internalCertificateId;
```

Comment:
But... these will have the same certificateId. I think you wanted to sort by date